pkg/cloud/aws/services/getters.go

```diff
@@ -0,0 +1,47 @@
+// Copyright © 2018 The Kubernetes Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package services
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsprovider/v1alpha1"
+	ec2svc "sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/services/ec2"
+	elbsvc "sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/services/elb"
+)
+
+// SDKGetter is the default getter based on the AWS SDK.
+type SDKGetter struct{}
+
+// Session returns a new AWS session.
+func (*SDKGetter) Session(clusterConfig *v1alpha1.AWSClusterProviderConfig) *session.Session {
+	return session.Must(session.NewSession(aws.NewConfig().WithRegion(clusterConfig.Region)))
+}
+
+// EC2 returns a new EC2 service client.
+func (*SDKGetter) EC2(session *session.Session) EC2Interface {
```

Comment:
feels a little bit weird to go through the hoop having to create a session then use it to create an EC2 service. Would it be easier for the user if `EC2` had the cluster config (or region) as a parameter without having to explicitly get a session every time?

Reply:
This Getter satisfier the Getter interface which is a combination of multiple interfaces, it can be further refactored if there is no need to separate them, what do you think?

Reply:
I guess my issue is with the Getter interface. I think that could be cleaned up a bit. Nothing to do on this PR though